swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -72,7 +47,19 @@ class Adapter {
   /// Network routes monitor.
   private var networkMonitor: NWPathMonitor?
 
-  /// Private queue used to synchronize access to `WireGuardAdapter` members.
+  /// Used to avoid path update callback cycles on iOS
+  #if os(iOS)
+    private var gateways: [Network.NWEndpoint] = []
+  #endif
+
+  /// Track our last fetched DNS resolvers to know whether to tell connlib they've updated
+  private var lastFetchedResolvers: [String] = []
```

Comment:
I think connlib already does that on main, although it doesn't filter out the sentinels correctly yet. Gabi and I each have different PRs open for that.